12-offer-encoding.md

```diff
@@ -229,6 +229,88 @@ The human-readable prefix for offers is `lno`.
     1. type: 22 (`offer_issuer_id`)
     2. data:
         * [`point`:`id`]
+    1. type: 24 (`offer_recurrence`)
+    2. data:
+        * [`byte`:`time_unit`]
+        * [`tu32`:`period`]
+    1. type: 25 (`offer_recurrence_paywindow`)
+    2. data:
+        * [`byte`:`proportional_amount`]
+        * [`u32`:`seconds_before`]
+        * [`tu32`:`seconds_after`]
+    1. type: 26 (`offer_recurrence_limit`)
+    2. data:
+        * [`tu32`:`max_period`]
+    1. type: 27 (`offer_recurrence_base`)
+    2. data:
+        * [`byte`:`start_any_period`]
+        * [`tu64`:`basetime`]
+
+### Recurrence
+
+Some offers are *periodic*, such as a subscription service or monthly dues, in that payment is expected to be repeated.  There are many different flavors of repetition, consider:
+
+* Payments due on the first of every month, for 6 months.
+* Payments due on every Monday, 1pm Pacific Standard Time.
+* Payments due once a year:
+   * which must be made on January 1st, or
+   * which are only valid if started January 1st 2021, or
+   * which if paid after January 1st you (over) pay the full rate first year, or
+   * which if paid after January 1st are paid pro-rata for the first year, or
+   * which repeat from whenever you made the first payment
+
+Thus, each offer containing a recurring payment has:
+1. A `time_unit` defining 0 (seconds), 1 (days), 2 (months), 3 (years).
```

Comment:
I understand the need for seconds, days and months but years seem redundant as 1 year is always 12 months.

Reply:
True.  I could argue that 4 years is better UX than 48 months, but the same argument would apply for minutes and hours.  I'll drop it!